onnxruntime/core/providers/openvino/qdq_transformations/qdq_stripping.cc

```diff
@@ -0,0 +1,699 @@
+
+  // If UInt16 Q, don't keep it
+  if (GetQDQDataType(q_node) == ONNX_NAMESPACE::TensorProto_DataType_UINT16) {
+    reason = SkipReason::Uint16QDQ;
+    return false;
+  }
+
+  if (op_type == "Conv" || op_type == "MatMul") {
+    // Conv and MatMul keep all Qs except if the target that succeeds it is Add/Mul/Div AND has any const init
+    return IsNextTargetNodeOfQValid(q_node, &target_node, src_graph, {"Add", "Mul", "Div"}, true);
+  } else if (op_type == "Add") {
+    // Add keeps all Qs
+    return true;
+  } else {
+    // Keep Q of an unsupported Op only if the target that succeeds it is a supported Op in this list
+    return IsNextTargetNodeOfQValid(q_node, &target_node, src_graph, {"Conv", "Add", "MatMul"}, false);
+  }
+}
+
+static bool HandleDoubleQDQ(onnxruntime::Graph& dst_graph, const onnxruntime::GraphViewer& src_graph,
+                            const NodeUnit& node_unit, std::set<std::string>& initializers_to_keep) {
+  int node_unit_input_edge_count = node_unit.InputEdgeCount();
+  int node_unit_output_edge_count = [&]() {
+    int count = 0;
+    for (auto it = node_unit.OutputEdgesBegin(); it != node_unit.OutputEdgesEnd(); ++it)
+      count += 1;
+    return count;
+  }();
+  bool edges_exist = node_unit_input_edge_count && node_unit_output_edge_count;
+
+  // Detect a conversion between quantized types (e.g., int16 to int8)
+  // in mixed-precision QDQ models. The pattern a standalone DQ followed by a standalone Q:
+  // Ex: DQ(int16 to float) -> QuantizeLinear(float to int8) ->
+  if (node_unit.OpType() == "QuantizeLinear" && edges_exist) {
+    const Node& q_node = node_unit.GetNode();
+    const Node& i_dq_node = *q_node.InputNodesBegin();
+    const Node& o_dq_node = *q_node.OutputNodesBegin();
+
+    if (i_dq_node.OpType() == "DequantizeLinear" && o_dq_node.OpType() == "DequantizeLinear") {
+      auto q_zero_point_dt = GetQDQDataType(&q_node);
+
+      // Can ignore if this Q is uint16 as it won't be consumed by any supported node
+      if (q_zero_point_dt != ONNX_NAMESPACE::TensorProto_DataType_UINT16 &&
+          !IsQDQSandwichedBetweenSoftmaxAndConvMatMulOps(&q_node)) {
+        // if it's unequal, then it's a conversion between quantized types.
+        AddNode(initializers_to_keep, src_graph, dst_graph, node_unit.GetNode());
+        return true;
+      }
+    }
+  }
+
+  // Keep int8 DQ/Qs in int16 -> int8
+  // Don't keep any QDQs in int8 -> int16
+  // Beginning of a converting QDQ pair. Check if previous Q is int8 and the following Q has a converting zp
+  if (node_unit.OpType() == "DequantizeLinear" && edges_exist) {
+    const Node& dq_node = node_unit.GetNode();
+    const Node& i_q_node = *dq_node.InputNodesBegin();
+    const Node& o_q_node = *dq_node.OutputNodesBegin();
+
+    if (i_q_node.OpType() == "QuantizeLinear" && o_q_node.OpType() == "QuantizeLinear") {
+      auto dq_zero_point_dt = GetQDQDataType(&dq_node);
+
+      if (dq_zero_point_dt != ONNX_NAMESPACE::TensorProto_DataType_UINT16 &&
```

Comment:
The comment above says "Keep int8 DQ/Qs in int16 -> int8", but if I'm not mistaken, this condition does the opposite. A int16 -> int8 conversion starts with a DequantizeLinear w/ int16 zp.

Reply:
Fixed comment to clarify what is done